### PR TITLE
Don't enforce a maven URL has 6 parts since one is optional

### PR DESCRIPTION
**Pull request by rowillia**

### CODE REVIEW

File: src/com/facebook/buck/file/MavenUrlDecoder.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public static URI toHttpUrl(Optional<String> mavenRepo, URI uri) {
 
     Matcher matcher = URL_PATTERN.matcher(uri.getSchemeSpecificPart());
 
-    if (!matcher.matches() || matcher.groupCount() != 6) {
+    if (!matcher.matches()) {
```
Comment (sdwilsh) [inline src/com/facebook/buck/file/MavenUrlDecoder.java]: I guess writing a test for this would be rather difficult, wouldn't it?

**Comment by sdwilsh** [comment]
@shs96c and I would like to see a test for this

**Comment by rowillia** [comment]
Fiiiiiiiiiiine....added a test

File: src/com/facebook/buck/file/MavenUrlDecoder.java (hunk 1 above)
Comment (rowillia) [inline src/com/facebook/buck/file/MavenUrlDecoder.java]: Added a test

**Pull request closed without merging**